### DISCUSSION TITLE
[SOLVED] If @clr/ui is being used in @cds/core, why not decouple the angular dependency?

### DISCUSSION THREAD

**Discussion by squidjam** [vmware-clarity/ng-clarity#670] [Ideas]
I mean, all I need is to get the CSS to correctly show my simplistic core-based web-components (like the header, or the dropdown) in a React project.

Wouldn't it be better that one could install the necessary CSS without having the additional load of having @angular v15 as a dependency?

It clashes with Storybook which requires @angular v16 to get StoryShots going (visual regression testing, it's cool, you'd love it). 

And it's not like it is ACTUALLY needing angular (the UI I mean)... It's just that the SCSS resides in the NG repo which, conversely, requires @angular.

Would it take a lot to create a simplistic UI repo for the UI that generates the needed CSS, clean of other dependencies?

Why keep it as is? 

I might not be understanding something.

**Comment by kevinbuhmann** [comment]
>If @clr/ui is being used in @cds/core

It's not.

>I mean, all I need is to get the CSS to correctly show my simplistic core-based web-components (like the header, or the dropdown) in a React project.
>
>Wouldn't it be better that one could install the necessary CSS without having the additional load of having https://github.com/angular v15 as a dependency?

If you are only using `@cds/core` (Clarity Core) web components, you don't need `@clr/ui` or Angular. The CSS for the Clarity Core web components is in the `@cds/core` package. There are some global styles needed. See here: https://core.clarity.design/get-started/developing

>Would it take a lot to create a simplistic UI repo for the UI that generates the needed CSS, clean of other dependencies?

The `@clr/ui` source exists in the ng-clarity repo alongside the Angular component library, but the package itself is just the CSS without any dependencies. See here: https://www.npmjs.com/package/@clr/ui?activeTab=code

>visual regression testing, it's cool, you'd love it

We actually have [visual regression tests on our ng-clarity components](https://github.com/vmware-clarity/ng-clarity/tree/main/tests/snapshots). It is cool, and we do love it. 💯

**Reply by squidjam** [reply]
Wait, so maybe I should be talking to the Storybook guys, because when I had @clr/ui (to use the heading CSS for the heading and dropdown components that aren't yet present in @cds/core yet) as I tried to install StoryShots, it gave me grief, and the message in NPM's error was pointing at @clr/ui.

Sure enough, uninstalling it, installing StoryShots and then installing @clr/ui did the trick.

OK, a hiccup then and nothing more.

Thanks for taking the time to read my comment and for answering it, I'm hoping to contribute to core soon-ish with some ideas.

Cheers.
